scripts/generate-locales.ts

```diff
@@ -0,0 +1,118 @@
+import ts from 'typescript';
+import path, { relative } from 'path';
+import { writeFile, mkdir, stat } from 'fs/promises';
+import { glob, chalk } from 'zx';
+
+const SRC_DIR = path.resolve('./src/lib/i18n/locales');
+const DEST_DIR = path.resolve('./static/i18n/locales');
+
+const logAndExit = (msg: string, logError: (msg: string) => void) => {
+  logError(chalk.red(msg));
+
+  if (logError === console.error) {
+    process.exit(0);
+  }
+};
+
+const relativeTo = (path: string) => {
+  return relative('./', path);
+};
+
+export const generateLocales = async (
+  src: string = SRC_DIR,
+  dest: string = DEST_DIR,
+  logError: (msg: string) => void = console.error,
+) => {
+  const pattern = path.join(src, '*/*.ts');
+  const filesToTranslate = await glob(pattern);
+
+  filesToTranslate.map(async (source) => {
+    const program = ts.createProgram([source], ts.getDefaultCompilerOptions());
+    const sourceFile = program.getSourceFile(source);
+    const checker = program.getTypeChecker();
+
+    if (!sourceFile) {
+      logAndExit(
+        `Error retrieving source file information for ${source}`,
+        logError,
+      );
+    }
+
+    const sourceFileSymbol = checker.getSymbolAtLocation(sourceFile!);
```

Comment:
Why the `!`?

Reply:
Telling the TypeScript compiler that we do have a defined `sourceFile` here, because we are checking for its truthiness in the lines above. https://github.com/temporalio/ui/pull/1362/files#diff-c113f7c7730c2f6cf15e9c61f2bb2be52ea8921825a0dd8f1aac12eed4f75a22R34-R39

I could refactor the code to avoid these assertions, but then I think it becomes slightly less readable.

Reply:
Actually I can just add some `return`'s so that the compiler is happy, will do that.